app/src/main/cpp/BrowserWorld.cpp

```diff
@@ -548,23 +548,22 @@ BrowserWorld::State::UpdateControllers(bool& aRelayoutWidgets) {
     if (controller.modelToggle)
       controller.modelToggle->ToggleAll(controller.mode == ControllerMode::Device);
 
+    // If we are rendering hands using spheres, update the joints' transforms here.
+    if (controller.handJointTransforms.size() > 0) {
```

Comment:
Why we don't need to check for the controller.handMesh now ?

Reply:
Before this PR, we were using the `meshJointTransforms` or `handJointTransforms` vectors to store the joint locations coming from the backend, depending on whether it was Oculus/Lynx (generic hand mesh) or Pico/A3 (spheres). That was confusing and more complex for no reason. At the end, both vectors ultimately store a vrb::Matrix.

This PR changes that by storing the joint locations in the Controller::meshJointTransforms (which is a vector of vrb::Matrix too), and then at render time, we just update the matrix of the spheres' vrb::Transform.

That simplifies the code, as you can see below, where we remove the condition of which array to index into to calculate left hand button action. This will also simplify things when we remove the spheres code-path, since we won't need to touch the rest of the code. 

I had this change already planned, as you can see a similar commit in PR#821 for XR_MSFT_hand_tracking_mesh (3rd commit in the series).

Reply:
I see, wouldn't be better then to get rid of one of the vectors ? it seems that we still use both; if they are going to eventually have the same values, why we need different vectors ?

Reply:
`handJointTransforms` is a vector of `vrb::Transform`, which is a node in the vrb scene graph. These nodes apply a transform to their children, which in this case are the vrb::Geometry for the spheres. So we need this vector to support rendering the spheres for Pico and A3 currently.

The vrb::Transform node has a vrb::Matrix member, which is what needs to be set. Previously, we were setting that member directly from the vector coming from the backend. With this PR, we first store the matrices in `meshJointTransforms`, then at render time we assign the matrices to the vrb::Transform.